server/src/main/java/com/flipkart/varadhi/db/VaradhiMetaStore.java

```diff
@@ -162,37 +164,41 @@ public int updateProject(Project project) {
 
     @Override
     public List<String> getVaradhiTopicNames(String projectName) {
-        String projectPrefixOfTopicName = projectName + RESOURCE_NAME_SEPARATOR;
+        String projectPrefixOfTopicName = projectName + NAME_SEPARATOR;
         ZNode znode = ZNode.OfEntityType(VARADHI_TOPIC);
         return zkMetaStore.listChildren(znode)
                 .stream()
                 .filter(name -> name.contains(projectPrefixOfTopicName))
-                .map(name -> name.split(RESOURCE_NAME_SEPARATOR)[1])
                 .collect(Collectors.toList());
     }
 
     @Override
-    public void createTopicResource(TopicResource resource) {
-        ZNode znode = ZNode.OfTopicResource(resource.getProject(), resource.getName());
-        zkMetaStore.createZNodeWithData(znode, resource);
-    }
-
-    @Override
-    public boolean checkTopicResourceExists(String topicResourceName, String projectName) {
-        ZNode znode = ZNode.OfTopicResource(projectName, topicResourceName);
-        return zkMetaStore.zkPathExist(znode);
-    }
-
-    @Override
-    public TopicResource getTopicResource(String topicResourceName, String projectName) {
-        ZNode znode = ZNode.OfTopicResource(projectName, topicResourceName);
-        return zkMetaStore.getZNodeDataAsPojo(znode, TopicResource.class);
+    public VaradhiTopic getVaradhiTopic(String topicResourceName, String projectName) {
+        String varadhiTopicName = getVaradhiTopicName(topicResourceName, projectName);
+        if("".equals(varadhiTopicName)) {
+            throw new ResourceNotFoundException(String.format(
+                    "Topic(%s) not found for the Project(%s).",
+                    topicResourceName,
+                    projectName
+            ));
+        }
```

Comment:
Shouldn't list all topics to get a varadhiTopic. VaradhiTopic name is convention based and same can be followed here. I think caller can construct the name and pass it down ?
Same would be applicable to other methods in this class.